plutus-scb/src/Cardano/Metadata/Client.hs

```diff
@@ -12,44 +12,47 @@
 import           Control.Monad.Freer       (Eff, LastMember, Member, type (~>), interpret, sendM)
 import           Control.Monad.Freer.Error (Error, throwError)
 import           Control.Monad.IO.Class    (MonadIO, liftIO)
-import           Data.Coerce               (coerce)
+import           Data.Coerce               (Coercible, coerce)
 import           Data.Proxy                (Proxy (Proxy))
 import           Network.HTTP.Types        (status404)
-import           Servant.Client            (ClientEnv, ClientError (FailureResponse), ClientM, client,
+import           Servant.Client            (Client, ClientEnv, ClientError (FailureResponse), ClientM, client,
                                             responseStatusCode, runClientM)
 import           Servant.Extra             (left, right)
 
+-- Maintainer's Note: The api can be flipped between 'AesonEncoding'
+-- and 'ExternalEncoding' depending on which metadata server we're
+-- using. See the notes on this, and 'coerce', in
+-- 'Cardano.Metadata.Types'.
+api :: Client ClientM (API 'AesonEncoding)
+api = client (Proxy @(API _))
+
 handleMetadataClient ::
        forall m effs.
        (LastMember m effs, MonadIO m, Member (Error MetadataError) effs)
     => ClientEnv
     -> Eff (MetadataEffect ': effs) ~> Eff effs
 handleMetadataClient clientEnv =
-        -- The api can be flipped between 'AesonEncoding' and
-        -- 'ExternalEncoding' depending on which metadata server
-        -- we're using. See the notes on this, and 'coerce', in
-        -- 'Cardano.Metadata.Types'.
-    let api = client (Proxy @(API 'ExternalEncoding))
-
-        (getProperties, getProperty) = (left . api, right . api)
-
-        runClient :: ClientM (Maybe a) -> Eff effs (Maybe a)
-        runClient action = do
+    let getProperties subject = left (left api subject)
+        getProperty subject = right (left api subject)
+        batchQuery = right api
+        runClient :: Coercible a b => a -> ClientM b -> Eff effs a
+        runClient def action = do
             result <- sendM $ liftIO $ runClientM action clientEnv
             case result of
-                Right value -> pure value
+                (Right value) -> pure $ coerce value
                 (Left (FailureResponse _ response))
-                    | responseStatusCode response == status404 -> pure Nothing
+                    | responseStatusCode response == status404 -> pure def
                 (Left err) -> throwError $ MetadataClientError err
      in interpret $ \case
-            GetProperties subject ->
-                coerce <$> runClient (getProperties subject)
+            GetProperties subject -> runClient Nothing $ getProperties subject
             GetProperty subject propertyKey ->
-                coerce <$> runClient (getProperty subject propertyKey)
+                runClient Nothing $ getProperty subject propertyKey
+            BatchQuery query -> runClient [] $ batchQuery query
```

Comment:
I guess it's too late to bring this up now but the 'http' way of doing things would be to return a 404 for all of these, rather than a 200 with an empty result. Not a big deal but it makes this API a little less accessible for other languages as you have to decode `Maybe` etc.

Reply:
Yeah, you're right. Fixed.